assets/js/base/hooks/customer/use-customer-data.ts

```diff
@@ -18,31 +26,39 @@ import { useStoreCart } from '../cart';
 /**
  * This is a custom hook for syncing customer address data (billing and shipping) with the server.
  */
-export const useCustomerData = () => {
+export const useCustomerData = (): {
+	billingData: CartResponseBillingAddress;
+	shippingAddress: CartResponseShippingAddress;
+	setBillingData: ( data: CartResponseBillingAddress ) => void;
+	setShippingAddress: ( data: CartResponseBillingAddress ) => void;
+} => {
 	const { updateCustomerData } = useDispatch( storeKey );
 	const { addErrorNotice, removeNotice } = useStoreNotices();
 
 	// Grab the initial values from the store cart hook.
 	const {
 		billingAddress: initialBillingAddress,
 		shippingAddress: initialShippingAddress,
-	} = useStoreCart();
+	}: CustomerData = useStoreCart();
 
 	// State of customer data is tracked here from this point, using the initial values from the useStoreCart hook.
-	const [ customerData, setCustomerData ] = useState( {
-		billingData: initialBillingAddress,
+	const [ customerData, setCustomerData ] = useState< CustomerData >( {
+		billingAddress: initialBillingAddress,
```

Comment:
In this hook only, I renamed `billingData` to `billingAddress`

Reply:
@senadir we should leave it as billingData if it contains the email and tel. Does it here?

Reply:
I believe we call it billingData in places because it contains email + tel, not just address. I think we should leave naming alone until reviewing this across the client.

Reply:
In our typedef and API response, billingAddress always contain email and tel, I can't find any instance in which billingAddress has only the address, yet we have two namings for it.
I can roll this back but just wanted to be consistent with the API and types.

Reply:
Updated this with some gymnastics, still not convinced this should be billingData.